pkg/crosscluster/logical/table_batch_handler_test.go

```diff
@@ -7,89 +7,60 @@ package logical
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
 	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/desctestutils"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
-	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
 )
 
-type eventBuilder struct {
-	descID descpb.ID
-}
-
-func newEventBuilder(
-	t *testing.T, runner *sqlutils.SQLRunner, s serverutils.ApplicationLayerInterface,
-) *eventBuilder {
-	runner.Exec(t, `
-		CREATE TABLE test_table (
-			id STRING PRIMARY KEY,
-			value STRING
-		);
-	`)
-
-	// Get the actual table descriptor
-	desc := desctestutils.TestingGetPublicTableDescriptor(s.DB(), s.Codec(), "defaultdb", "test_table")
-
-	return &eventBuilder{
-		descID: desc.GetID(),
-	}
-}
-
-func (e *eventBuilder) makeInsert(id string, value string, timestamp hlc.Timestamp) decodedEvent {
-	return decodedEvent{
-		dstDescID:       e.descID,
-		isDelete:        false,
-		originTimestamp: timestamp,
-		row:             tree.Datums{tree.NewDString(id), tree.NewDString(value)},
-	}
-}
-
-func (e *eventBuilder) makeUpdate(
-	id string, old, new string, timestamp hlc.Timestamp,
-) decodedEvent {
-	return decodedEvent{
-		dstDescID:       e.descID,
-		isDelete:        false,
-		originTimestamp: timestamp,
-		row:             tree.Datums{tree.NewDString(id), tree.NewDString(new)},
-		prevRow:         tree.Datums{tree.NewDString(id), tree.NewDString(old)},
-	}
-}
-
-func (e *eventBuilder) makeDelete(id string, value string, timestamp hlc.Timestamp) decodedEvent {
-	return decodedEvent{
-		dstDescID:       e.descID,
-		isDelete:        true,
-		originTimestamp: timestamp,
-		row:             tree.Datums{tree.NewDString(id), tree.DNull},
-		prevRow:         tree.Datums{tree.NewDString(id), tree.NewDString(value)},
-	}
-}
-
-func newTestTableHandler(
-	t *testing.T, descID descpb.ID, s serverutils.ApplicationLayerInterface,
-) *tableHandler {
-	sd := sql.NewInternalSessionData(context.Background(), s.ClusterSettings(), "" /* opName */)
-	handler, err := newTableHandler(context.Background(), descID, s.InternalDB().(descs.DB), s.Codec(), sd, 1337, s.LeaseManager().(*lease.Manager), s.ClusterSettings())
+func newTestBatchHandler(
```

Comment:
nit: since this creates a crudBatchHandler, could you rename this helper to `newCrudBatchHandler`?